### PR TITLE
Master.lossfunctions

### PR DESCRIPTION
**Pull request by Giounona**
Focal losses and addressed comments on master.lossgt previous branch

### CODE REVIEW

File: src/optimization/loss.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
+import numpy as np
+from torch.nn.modules.loss import _Loss
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.autograd import Variable
+
+
+class Focaloss_experimental(torch.nn.Module):
```
Comment (JakeForsey) [inline src/optimization/loss.py]: Can we keep just one of these?

Delete this one

File: train.py [hunk 2]
```diff
@@ -53,26 +60,51 @@ def run_experiment(
...
 
         with APTOSMonitor(experiment, cv_iteration) as monitor:
             LOGGER.info(f'tensorboard --logdir "{monitor._summary_writer.log_dir}"')
 
-            test_size = experiment.test_size()
-            train_ds, test_ds = torch_random_split(
-                dataset,
-                [round((1 - test_size) * len(dataset)), round(test_size * len(dataset))]
-            )
+            if experiment.dataset_split()=="stratified":
```
Comment (JakeForsey) [inline train.py]: Lets just make this split the only possible split so that we can get rid of Experiment.dataset_split()

File: src/optimization/loss.py [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+        if self.size_average:
+            loss = loss.mean()
+        else:
+            loss = loss.sum()
+        return loss
+
+
+
+class FocalLoss2(nn.Module):
```
Comment (JakeForsey) [inline src/optimization/loss.py]: Delete this one too
